packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1762,7 +1765,18 @@ export class SnapController extends BaseController<
    * @param promise - The promise to await.
    * @returns The result of the promise or rejects if the promise times out.
    */
-  private async _executeWithTimeout(promise: Promise<unknown>) {
+  private async _executeWithTimeout(snapId: SnapId, promise: Promise<unknown>) {
+    const isLongRunning = this.messagingSystem.call(
+      'PermissionController:hasPermission',
+      snapId,
+      LONG_RUNNING_PERMISSION,
+    );
+
+    // Long running snaps have timeouts disabled
+    if (isLongRunning) {
+      return promise;
+    }
```

Comment:
I'm thinking should probably just leave this alone for now with the intention to eventually support some kind of "long-running worker" API and add some kind of timeout here at that point. We need to make sure to warn about this development in big red letters in the docs, but other than that, this is OK.

Reply:
👍